agent/api/task/task.go

```diff
@@ -2548,3 +2556,62 @@ func getDomainForPartition(region string) string {
 	}
 	return partition.DNSSuffix()
 }
+
+func (task *Task) requireEnvfiles() bool {
+	for _, container := range task.Containers {
+		if container.ShouldCreateWithEnvFiles() {
+			return true
+		}
+	}
+	return false
+}
+
+func (task *Task) initializeEnvfilesResource(config *config.Config, credentialsManager credentials.Manager, ) error {
+
+	for _, container := range task.Containers {
+		envfileResource, err := envFiles.NewEnvironmentFileResource(config.Cluster, task.Arn, config.AWSRegion, config.DataDir,
+			container.EnvironmentFiles, credentialsManager, task.ExecutionCredentialsID)
+		if err != nil {
```

Comment:
Since each container has their own task resource, you may want to have container name or other container identifier in the error message so people can understand which one is causing the error.

Reply:
good point - also added check for containers having env files before creating a resource. instead of checking all containers and then checking again.